### DISCUSSION TITLE
Different forms to reset the password

### DISCUSSION THREAD

**Discussion by Halvanhelv** [janko/rodauth-rails#207] [Q&A] [answered]
Hi! i have a few questions but they are related, need advice/examples of solutions

1) I have a profile page and it has a button to reset the password (I use it if a person wants to change the mail but registered through oauth and does not have a password) 

I do not want to force an authenticated user to enter his login to reset, so I changed the standard form a bit and substituted the login myself 

```ruby
<%= turbo_frame_tag "reset_password_button" do %>
<%= form_with url: rodauth.reset_password_request_path, method: :post, data: { turbo: false }, class: "w-full max-w-sm" do |form| %>
    <%= form.hidden_field rodauth.login_param, value: current_user.email %>
  <%= form.submit rodauth.reset_password_request_button %>
<% end %>
<% end %>
```

is it ok to do this or is there a better solution?

2) I want to use password recovery form in two different places and it should look different, how can it be done if I use this code to return html for one form if it is filled out incorrectly but the second form needs completely different html, how to distinguish which of the two forms the request came from?

Or maybe there is a completely different approach for this kind of task and it is worth redoing?

```ruby
    reset_password_request_view {
      rails_render(
        turbo_stream: [
          rails_controller_instance.turbo_stream.replace('reset_password_button', partial: 'profile/reset_password'),
          rails_controller_instance.turbo_flashes
        ]
      )
    } 
```

**Comment by janko** [comment, marked as answer]
Hello 👋🏻 

1. Yes, that seems like a good solution to me 👍🏻 
2. You could add a custom form field, and then check it with e.g. `if param("source") == "profile"` in a configuration method block.

**Reply by Halvanhelv** [reply]
thanks, now it works
